tests/5.1/order/test_taskloop_order_reproducible_device.c

```diff
@@ -0,0 +1,54 @@
+//===------------ test_target_taskloop_order_reproducible_device.c ----------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test checks that the order(reproducible:concurrent) clause is properly handled.
+// Leverages the previous array item to determine if the same threads executed
+// in the same order over the array. If both arrays match the test passes, 
+// if not it fails as the reproducible clause failed.
+//
+////===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+	int errors = 0;
+	int x[N];
+	int y[N];
+
+	for (int i = 0; i < N; i++) {
+		x[i] = i;
+		y[i] = i;
+	}
+
+	OMPVV_TEST_OFFLOADING;
+
+	#pragma omp target map(tofrom: x,y)
+	{
+
+		#pragma omp parallel
+		{
+		   #pragma omp taskloop order(reproducible:concurrent)
```

Comment:
This testcase fails with GCC as follows:
```
error: ‘order’ is not valid for ‘#pragma omp taskloop’
```

If I look at OpenMP 5.2, I see:
> 12.6 `taskloop` Construct
> ...
> **Clauses**
> allocate, collapse, default, final, firstprivate, grainsize, if, in_reduction, lastprivate, mergeable, nogroup, num_tasks, priority,1 private, reduction, shared, untied

which does not list  the `order` clause. That clause in turn is only permitted for:

> 10.3 `order` Clause
> ...
> **Directives**
> distribute, do, for, loop, simd

Unsurprisingly, OpenMP 5.1 also does not permit it: https://www.openmp.org/spec-html/5.1/openmpsu55.html#x81-910002.12.2